builder/proxmox/iso/config.go

```diff
@@ -23,6 +27,12 @@ type Config struct {
 	// `local:iso/Fedora-Server-dvd-x86_64-29-1.2.iso`.
 	// Either `iso_file` OR `iso_url` must be specifed.
 	ISOFile string `mapstructure:"iso_file"`
+	// Bus type and bus index that the ISO will be mounted on. Can be `ideX`,
+	// `sataX` or `scsiX`.
+	// For `ide` the bus index ranges from 0 to 3, for `sata` from 0 to 5 and for
+	// `scsi` from 0 to 30.
+	// Defaults to `ide2`
+	ISODevice string `mapstructure:"iso_device"`
```

Comment:
I wonder if this should just be an `ISODeviceType`, so they are all appended to the list of the appropriate type without needing to give it an index?

Reply:
Yeah regarding disks I can see a reason why we'd want to keep the order in which we enumerate them (even when modern OSes don't access their devices by the order they're discovered but through UUID), regarding cdrom devices, I believe they can fill-in the remaining slots, that would be likely be simpler and probably won't break existing configs.
What do you think?

Reply:
I tracked down the original additional_iso_files issue https://github.com/hashicorp/packer/issues/7950 and PR https://github.com/hashicorp/packer/pull/9653 that brought the logic into the builder to understand why a device and index are specified. It looks like the builder at the time only accepted one set of iso device config (what is now the proxmox-iso builder) and additional_iso_files was added in.

It seems the device and index weren't explicit requirements in the feature request, so we could look at removing the index attribute for all ISOs and just enable the assignment of a device type. Looking at https://github.com/hashicorp/packer/pull/9653#discussion_r467550534 we've ended up implementing the second proposed option for disks to meet the new API struct format anyway so we would just be folding in ISO devices.

It would certainly simplify aspects of this PR 😉 however this would be a breaking change for existing packer configs where an additional_iso_files block has a device field value.

I'm happy to have a go at updating everything to use device type assignments without indexes.. with the above information do you think this is a viable way forward?